### PR TITLE
feat(cli): New add method commands

### LINKED ISSUE #955: Create commands to add reducers/projections to existing entities/read-models
It'd be nice to have the following commands:

* `boost add:reducer --entity MyEntity --event MyEvent`
* `boost add:projection --read-model MyReadModel --entity MyEntity:id`

### PR DESCRIPTION
**Pull request by Fecony**
## Description
This PR is adding new `boost:add` command that will help generating methods in Entity and ReadModel files ✨

generated_reducer_example

### Add Projection to ReadModel
Example command: 
```bash 
$ boost add:projection --read-model PostReadModel --entity Post:id
```

Will generate projection method in `post-read-model.ts` file `PostReadModel` class:
```ts
@Projects(Post, 'id')
public static projectPost(entity: Post, currentPostReadModel?: PostReadModel): ProjectionResult<PostReadModel> {
  return /* NEW PostReadModel HERE */
}
```

### Add reducer to Entity
Example command: 
```bash 
$ boost add:reducer --entity Post --event PostUpdated
```

Will generate reducer method in `post.ts` file `Post` class:
```ts
@Reduces(PostUpdated)
public static reducePostUpdated(event: PostUpdated, currentPost?: Post): Post {
  return /* NEW Post HERE */
}
```
> Note: you can pass multiple events to reducer command, it will generate reducer method for each event

#### Each command will fix missing imports in source file
Nice thing from ts-morph is, if developer has set `paths` in tsconfig file `compilerOptions`, it will generate import with corresponding path
```ts
import { PostUpdated } from '@events/nested/path/post-updated'
```
## Changes
- new `boost:add projection` command
- new `boost:add reducer` command
- new `fileNameWithExtension` helper function
- new `parseProjectionField` parse function for single projection
- new `HasProjection` type for single projection
- `ts-morph` dependency
- new `services/method-generator.ts` file
  - new `getResourceSourceFile` function to search for source file
  - new `generateReducers` function to generate reducers method declarations for `boost:add reducer` command
  - new `generateProjection` function to generate projection method declaration for `boost:add projection` command
- new tests
  - fixed code style in `services/generator/target/parsing.test.ts` file

## Checks
- [x] Project Builds
- [x] Project passes tests and checks
- [ ] Updated documentation accordingly
 
## Additional information
Closes #955

### CODE REVIEW

File: packages/cli/src/services/method-generator.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      {
+        name: `current${readModel}`,
+        type: readModel,
+        hasQuestionToken: true,
+      },
+    ],
+    returnType: `ProjectionResult<${readModel}>`,
+    statements: [`return /* NEW ${readModel} HERE */`],
+  }
+}
```
Comment (javiertoledo) [inline packages/cli/src/services/method-generator.ts:83]: Interesting approach!

**Review by javiertoledo — APPROVED** [review]
I loved this feature and how you implemented it, thanks @Fecony!

**Comment by Fecony** [comment]
Pulled latest main branch changes to keep PR on track 🚀 
After that, however, the `package-lock` file changes seem to have become even larger, although everything still seems to work...

Anyway let me know if it generated something weird or something else, if there's anything I should do to fix it 🐙

**Comment by Fecony** [comment]
Applied latest main branch changes to keep PR on track ✅

**Comment by Fecony** [comment]
Applied latest main branch changes to keep PR on track 🐙 ✅

**Comment by Fecony** [comment]
Applied latest main branch changes to keep PR on track 🐙 ✅
- fixed small style issue in test file

**Comment by Fecony** [comment]
Re-applied latest main branch changes to keep PR on track 🐙 ✅

- fixed small style issue in test files
- fixed package/-lock file issues from last merge

**Review by rdiaz82 — APPROVED** [review]
LGTM

**Comment by gonzalogarciajaubert** [comment]
Thanks a lot @Fecony ! I created a new PR https://github.com/boostercloud/booster/pull/1032 to run the integration tests and merge your commits. I will close this PR

**Pull request closed without merging**